static/js/browser_history.js

```diff
@@ -40,7 +40,9 @@ export function update_web_public_hash(hash) {
 }
 
 export function save_old_hash() {
-    state.old_hash = window.location.hash;
+    const current_hash = window.location.hash;
+    state.old_hash = current_hash;
+    update_web_public_hash(current_hash);
```

Comment:
Can you extend the commit message to explain why editing this function will catch all places where the hash changes not via `hashchanged`?  It's not obvious to me.

Reply:
This is actually called via hashchanged. The change below is the one that was actually missing. It was not being called since we were using `pushState` there.

Reply:
OK but this is still petty confusing; we're now calling `update_web_public_hash` twice:
```
   ...
   # CALLING VIA save_old_hash
    const was_internal_change = browser_history.save_old_hash();                                         
    if (was_internal_change) {                                                                           
        return undefined;                                                                                
    }                                                                                                    
                                                                                                         
    // TODO: Migrate the `#reload` syntax to use slashes as separators                                   
    // so that this can be part of the main switch statement.                                            
    if (window.location.hash.startsWith("#reload")) {                                                    
        // We don't want to change narrow if app is undergoing reload.                                   
        return undefined;                                                                                
    }                                                                                                    
                                                                                                         
   # CALLING directly here
    const is_hash_web_public_compatible = browser_history.update_web_public_hash(current_hash);          
```

Reply:
Ah! Sorry for the silly mistake.